### DISCUSSION TITLE
State of the art, related work and future work for probabilistic diagnosis

### DISCUSSION THREAD

**Discussion by pablocnds** [Fundacion-de-Neurociencias/neurodiagnoses#3] [General]
Here we can discuss references for related work that can help shape the probabilistic diagnosis element of this project and understand necessities and challenges. In this first post I present an overall summarized view of the field and some references.

As the search advances, we can compile it all in a .md file in the repository.

First, I've used tools like Gemini's Deep Research and ScholarGPT to summarize the current state of the field. Here I compiled some takeovers I consider most relevant for this project:

## Takeovers from the field
**Data challenges**: A major problem is the heterogeneity in the data sources. Also quantity, quality and limited access to data. We might have to work with relatively small datasets.
**Interpretability and explainability** are of major importance for patient diagnosis. Desirably, it should be possible to explain how the model arrives to its predictions.
**State of the art accuracy**: Current models seem to already be very accurate (more so than humans). Most seem to not use multimodal data however. Apparently there are improvements to be made in robustness against data heterogeneity and early diagnosis.
**Longitudinal data**: Some models leverage longitudinal data for the predictions.
**Open Source concerns**: There might be possible legal issues in making open source models that utilize sensitive patient data.

Source: you can find here a [9-page report from Gemini's Deep Research](https://docs.google.com/document/d/e/2PACX-1vTeXJI452Lx93EuE1pf50iI-VyhucgXf-8Ex92f9qGK-HT-qzoLQLcsQ76tPH3Rs6jctPKQXNRS_sKu/pub).

## Study of existing models and pipelines
In the comments bellow I analyze a number of sources to find and compile common techniques for preprocessing and design of model architectures.

**Comment by manuelmenendezgonzalez** [comment]
This recent paper is very relevant because it describes a system for the categorization of biomarkers based on the primary mechanism on which they report. https://alz-journals.onlinelibrary.wiley.com/doi/10.1002/trc2.70050
We have extended this categorization to other conditions. See "Biomarker Clasifier" https://github.com/Fundacion-de-Neurociencias/neurodiagnoses/blob/main/models/biomarker_classifier.py

**Comment by pablocnds** [comment]
### Pre-trained models vs from scratch

As I can see in [1,2,3] pre-trained 2D CNN models can be a good approach for creating an Alzheimer's Disease predictor from 2D slices of MRI, which is good to tackle the relatively limited amount of data and requires less computational resources. This presumably can be generalized to other neurodegenerative diseases.

ResNet18 seems to be a good candidate as it is accurate [2] and compact.

### Data augmentation

According to [4], data augmentation might not be necessary with pre-trained models in MRI data. But other sources still use it, so I would consider both approaches.

Edit: In [5] they find that, at least with their method, data augmentation improves accuracy, class imbalance and overfitting.
Edit2: More research leads me to think that data augmentation is simply better. Most studies use it without question.

### Sources

1. Khojaste-Sarakhsi M, Haghighi SS, Ghomi SMTF, Marchiori E. Deep learning for Alzheimer's disease diagnosis: A survey. Artif Intell Med. 2022;130:102332. doi:10.1016/j.artmed.2022.102332
2. Benbakreti, Soumia & Samir, Benbakreti & Kadda, Benyahia & Abdelkader, Khobzaoui. (2024). Pretrained models for classification of dementia degree and treatment of Alzheimer’s disease. 2024. 135-138. 
3. Sato R, Iwamoto Y, Cho K, Kang D-Y, Chen Y-W. In: Comparison of CNN models with different plane images and their combinations for classification of Alzheimer's disease using PET images. Singapore: Springer; 2019. p. 169–77. https://doi.org/10.1007/978-981-13-8566-7_36.
4. Benbakreti, S., Benouis, M., Roumane, A. et al.: Impact of the data augmentation on the detection of brain tumor from MRI images based on CNN and pretrained models, In: Multimed Tools Appl,2023.
5. S. Afzal et al., "A Data Augmentation-Based Framework to Handle Class Imbalance Problem for Alzheimer’s Stage Detection," in IEEE Access, vol. 7, pp. 115528-115539, 2019, doi: 10.1109/ACCESS.2019.2932786.

**Comment by pablocnds** [comment]
### Preprocessing is important!

In my testing I tried a naive method of taking random resampled 2D slices from the ADNI1 dataset images and feeding it directly to a pre-trained 2D ResNet18. Unsurprisingly the accuracy was bad.

I checked the literature to see what preprocessing is usually done **for the ADNI1 dataset**, which is already standardized and has some preprocessing. From sources [6,7,8,9] I extracted the following techniques:

- Resampling.
- Skull Stripping using e.g. **FSL’s Brain Extraction Tool (BET)** -- Which is free to use.
- Spatial Normalization using e.g. **FSL's FLIRT** (MNI152 template) -- Also free.
- Maybe intensity normalization? Not mentioned on all papers, but for example [9] uses z-score normalization.
- Selective 2D slices: can be one slilce, a few, or many per volume.
  - In [6], for each axis pick the slice with largest relative area (total of 3 slices per volume).
  - In [7], within the coronal plane, 40 center slices per data sample. For each sample the 3 channels are filled as follows: the original slice, with the 3rd and 7th contiguous slices (this is meant to encode volume in each sample). Then the slices are randomly cropped in square sections.
  - In [8], the 60 most centered slices are chosen for all 3 axis. Then they manually choose the best quality ones.
  - [9] doesn't specify, but like [7] and [8] they used several slices as independent data samples.

### Dataset combination

Sometimes different datasets and collections are combined, but not necessarily. For example [8] uses only the "ADNI1 Complete 1Yr 1.5T" collection, while [9] combines “complete 2-year 1.5T” and “complete 3-year 1.5T” ADNI sets. I've also seen others combining sets from different initiatives.

### Sources

**Note**: These are just any examples I found that explain their pre-procesing for ADNI1. These sources might not be the most comprehensive.

6. Angkoso, C. V., Tjahyaningtijas, H. P. A., Purnama, I. K. E., & Purnomo, M. H. (2022). Multiplane convolutional neural network (Mp‐CNN) for Alzheimer’s disease classification. International Journal of Intelligent Engineering and Systems, 15(1), 329–340. https://doi.org/10.22266/ijies2022.0228.30
7. Pereira, M. E. de C. (2019). An extended‐2D CNN approach for diagnosis of Alzheimer’s disease through structural MRI [Master’s thesis, Universidade Estadual de Campinas, Faculdade de Engenharia Elétrica e de Computação].
8. Hassan, N., Miah, A. S. M., Suzuki, K., Okuyama, Y., & Shin, J. (2025). Stacked CNN‐based multichannel attention networks for Alzheimer disease detection. *Scientific Reports, 15*, 5815. https://doi.org/10.1038/s41598-025-85703-x
9. Heising, L., & Angelopoulos, S. (2022). *Operationalising fairness in medical AI adoption: Detection of early Alzheimer’s disease with 2D CNN*. BMJ Health Care Informat, 29, e100485. https://doi.org/10.1136/bmjhci-2021-100485

**Reply by manuelmenendezgonzalez** [reply]
**Update on Neuroimage Preprocessing: Integrating DeepPrep**

I’d like to share an update on our progress with the preprocessing pipeline. We are currently trying to integrate DeepPrep into Neurodiagnoses to accelerate and improve the robustness of our neuroimaging preprocessing step.

What we’ve done so far:

Customized Dockerfile:
We modified the original DeepPrep Dockerfile to use a public URL instead of “localhost” for downloading certain dependencies (e.g., FSL 6.0.5.1). Additionally, we commented out the sections for AFNI and ANTs that were causing errors, as those components are optional for our current use.

Building the Image:
We have faced difficulties generating the image at Docker so we could not complete the process. We have opened an [issue](https://github.com/Fundacion-de-Neurociencias/neurodiagnoses/issues).

Container Setup:
The container is configured to mount our data directories (BIDS dataset, results, etc.) and run the DeepPrep pipeline. This integration is aimed at ensuring that our preprocessing step can handle large volumes of data in a scalable and reproducible way.

Next Steps:
Complete the integration.
I’m looking forward to your suggestions and feedback on this integration. Thanks for your support!

**Comment by simonprovost** [comment]
Hi folks!

Sharing https://github.com/simonprovost/scikit-longitudinal which might be of interest :)

Paper: https://joss.theoj.org/papers/10.21105/joss.08481

Cheers!

**Reply by manuelmenendezgonzalez** [reply]
Thank you for sharing.
scikit-longitudinal seems very interesting for our platform neurodiagnoses.com
We have tried using it, but it seems to require Python 3.10-3.12, and the whole neurodiagnoses is using Python 3.13. Is there any update to Python 3.13 expected soon?

**Reply by simonprovost** [reply]
My pleasure! Thanks so much for having had it thinking fitting in one of your projects! Neurodiagnoses seem very interesting indeed!

~~Yeah, this was a limitation a week ago already; it was only under the 3.9 supported version due to one of our dependencies hardly updating their algorithm (Deep-Forest) too slowly — We forked it out to create a community-based version of Deep Forest supporting 3.10 to 3.12; I am working on 3.13 and 3.14, but there are NumPy-related difficulties; I'll try to have that done soon, but no promises as of yet (but I am trying :))! Can't you guys work under 3.12? Just curious! I believe not but worth asking~~

Wishing you a lovely weekend ahead :) 

Cheers,

**Reply by simonprovost** [reply]
Hi @manuelmenendezgonzalez and team

Hope all is well,

Good news :) `Sklong` for _py313_ is out 🥳

I had to make sure one of our community forked versions of `Deep Forest` is [building wheels up to Python 3.13](https://github.com/simonprovost/deep_forest_py310/pull/2) and then updated Sklong accordingly (See PR https://github.com/simonprovost/scikit-longitudinal/pull/76). Note that it'll be tough to go up to _py314_ due to some discrepancies between `numpy`, `cython` (and `Sklearn` accordingly).

Hope to see `Sklong` through [Neurodiagnoses](https://neurodiagnoses.com), so that I can add it in the [Community Hub](https://scikit-longitudinal.readthedocs.io/latest/community-hub/) related projects of `Sklong` ! I have explored more in depth, it's amazing what you guys are onto! If you guys publish other papers on that project, and used Sklong, hope to see Sklong paper cited, it really helps as you may know :) !

_––Quick credit for the request––_

<img width="922" height="93" alt="Screenshot 2025-12-01 at 20 12 12" src="https://github.com/user-attachments/assets/e48d0c2c-11d3-4f24-86f5-693f8f24812d" />

Cheers

**Comment by manuelmenendezgonzalez** [comment]
Hi,
That's a great and quick job from your side! Thank you very much. We have now installed scikit-longitudinal on neurodiagnoses.com and we will be testing it soon. 
Of course, we will be citing the
 Sklong paper in our published research. 
 Thank you!

**Reply by simonprovost** [reply]
Thanks for your kind words @manuelmenendezgonzalez ! 

Good luck with everything! You guys let me know if you have any question.

An update of the documentation is incoming, to be easier to understand the lib :)

Cheers
